ReactNativeDeferredTemplate/ios/ReactNativeDeferredTemplate/AppDelegate.m

```diff
@@ -0,0 +1,175 @@
+    }];
+    return YES;
+}
+
+- (void)registerForRemotePushNotifications {
+    [[UNUserNotificationCenter currentNotificationCenter] requestAuthorizationWithOptions:(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge) completionHandler:^(BOOL granted, NSError * _Nullable error) {
+        if (granted) {
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [[SFPushNotificationManager sharedInstance] registerForRemoteNotifications];
+             });
+        }
+
+        if (error) {
+            [SFLogger e:[self class] format:@"Push notification authorization error: %@", error];
+        }
+    }];
+}
+
+- (void)customizeLoginView {
+    SFSDKLoginViewControllerConfig *loginViewConfig = [[SFSDKLoginViewControllerConfig  alloc] init];
+    // Set showSettingsIcon to NO if you want to hide the settings icon on the nav bar
+    loginViewConfig.showSettingsIcon = YES;
+    // Set showNavBar to NO if you want to hide the top bar
+    loginViewConfig.showNavbar = YES;
+    loginViewConfig.navBarColor = [UIColor colorWithRed:0.051 green:0.765 blue:0.733 alpha:1.0];
+    loginViewConfig.navBarTitleColor = [UIColor whiteColor];
+    loginViewConfig.navBarFont = [UIFont fontWithName:@"Helvetica" size:16.0];
+    [SFUserAccountManager sharedInstance].loginViewControllerConfig = loginViewConfig;
+}
+
+- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken
+{
+    // Uncomment the code below to register your device token with the push notification manager
+//    [self didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
+}
+
+- (void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+    [[SFPushNotificationManager sharedInstance] didRegisterForRemoteNotificationsWithDeviceToken:deviceToken];
+    if ([SFUserAccountManager sharedInstance].currentUser.credentials.accessToken != nil) {
+     [[SFPushNotificationManager sharedInstance] registerSalesforceNotificationsWithCompletionBlock:nil failBlock:nil];
+    }
+}
+
+- (void)application:(UIApplication *)application didFailToRegisterForRemoteNotificationsWithError:(NSError *)error
+{
+    // Respond to any push notification registration errors here.
+}
+
+- (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options
+{
+    // Uncomment following block to enable IDP Login flow
+//    return [self enableIDPLoginFlowForURL:url options:options];
+    
+    return NO;
+}
+
+- (BOOL)enableIDPLoginFlowForURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
+     return [[SFUserAccountManager sharedInstance] handleIDPAuthenticationResponse:url options:options];
+}
+#pragma mark - Private methods
+
+- (void)initializeAppViewState
+{
+    if (![NSThread isMainThread]) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self initializeAppViewState];
```

Comment:
`strong-weak`?

Reply:
It's in all our templates ;-)
I don't think we need to worry about a retain cycle here: AppDelegate instance should exist for the duration of the life of the application.